core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -107,6 +107,30 @@ public static MetricsConfig forPositionDelete(Table table) {
     return new MetricsConfig(columnModes.build(), defaultMode);
   }
 
+  /**
+   * Creates a metrics config for a position delete file.
+   *
+   * @param props table configuration
+   */
+  public static MetricsConfig forPositionDelete(Map<String, String> props) {
```

Comment:
should we use the method from line 91?
```
 public static MetricsConfig forPositionDelete(Table table) {
```

Flink write properties allows override of table properties for write (e.g. compression configs). Do we have such use cases for metrics config?

I am asking because the current `fromProperties` method (for data files) is deprecated in favor of the method `forTable`.
```
  /**
   * Creates a metrics config from table configuration.
   *
   * @param props table configuration
   * @deprecated use {@link MetricsConfig#forTable(Table)}
   */
  @Deprecated
  public static MetricsConfig fromProperties(Map<String, String> props) {
```

Reply:
At least, we can avoid code duplication by extracting common code btw these two methods, if we do need this new variant.

Reply:
Extraction done.

Reply:
@chenjunjiedada what do you think of using the existing API? I know it would require changes in `FlinkAppenderFactory` to pass in the `Table` object.
```
public static MetricsConfig forPositionDelete(Table table) {
```

The benefit is that position delete file will get the consistent behavior as data file. unless we are saying position delete has its own schema, shared code path doesn't really make sense.
```
  private static MetricsConfig from(Map<String, String> props, Schema schema, SortOrder order) {
    ...
  }
```

Reply:
Actually, positions delete does NOT share the same behavior as data files, `SortedPosDeleteWriter` sorts the content.

What do you mean by has its own schema? Position deletes can have two schemas, one is [path, pos] and another is [path, pos, row]. Currently, it writes [path, pos] in `BaseTaskWriter`.

Reply:
@chenjunjiedada hmm. I think we should revert the last commit. if we are going to pass in the `Table` object, it should be non-null. Then we can consistently use `forTable` everywhere.

I would also be open with the `forProperties` approach, as you were saying that schema and sort order are for data rows/files.  I had some reservations because `MetricConfig` deprecated `forProperties` for data rows/files. It is a little weird to introduce it back for position delete. Flink can use `forPositionDelete(Table table)` API by passing a valid `Table` object to `FlinkAppenderFactory`.